android-metadata-generator/src/src/com/telerik/metadata/ClassRepo.java

```diff
@@ -18,7 +18,15 @@ public static void addToCache(ClassMapProvider classMapProvider) {
                 if (clazz != null) {
                     String errMsg = "Class " + className + " conflict: "
                                     + classMapProvider.getPath() + " and " + cachedProvider.getPath();
-                    throw new IllegalArgumentException(errMsg);
+
+                    // TODO: Pete: throwing here means users can't add their own apache http libraries,
+                    // because there are some left-overs in the android sdk which will coincide
+                    // and thus cause the generator to crash
+
+                    // Should we worry about this? If classes would crash, that would happen during the compilation step
+                    // which occurs before the metadata generation
+
+//                    throw new IllegalArgumentException(errMsg);
```

Comment:
We needed this check when metadata was generated on the basis of `.dex` files. It's now obsolete because, as you said, the compilation task will take care of providing an understandable error for the user, so we don't need to.
You can remove the comment and leave the code as it is, without the throw statement.